platform/entanglement/test/services/LinkServiceSpec.js

```diff
@@ -81,6 +88,23 @@ define(
                     expect(validate()).toBe(false);
                 });
 
+                it("does not allow parents that contains object", function () {
+                    object.id = 'abc';
+                    parentCandidate.id = 'xyz';
+                    parentCandidate.model.composition = ['abc'];
+                    expect(validate()).toBe(false);
+                });
+
+                it("does not allow parents without composition", function () {
```

Comment:
In reality, this parent object wouldn't have a composition capability. (see CompositionCapability.appliesTo).  I think that's a better test in this case.